draft-dekater-scion-dataplane.md

```diff
@@ -246,9 +250,23 @@ The full forwarding process for a packet transiting an intermediate AS consists
 3. The SCION router maps the egress interface ID in the current hop field of the SCION header to the destination "intra-protocol" address of the egress border router (where "intra-protocol" is the intra-domain forwarding protocol, e.g., MPLS or IP).
 4. The packet is forwarded within the AS by routers and switches based on the "intra-protocol" header.
 5. Upon receiving the packet, the SCION egress router strips off the "intra-protocol" header, again validates and updates the SCION header, and forwards the packet to the neighboring SCION router.
+6. The last SCION router on the path forwards the packet to the packet's destination endpoint indicated by the field `DstHostAddr` of [the Address Header](#address-header).
+
+### Configuration
+
+Border routers require mappings from SCION  interface IDs to underlay addresses. Such information must be supplied to each router in an out of band fashion (e.g in a configuration file). For each link to a neighbor, these values must be configured:
+
+- Local interface ID
+- Neighbor type (core, parent, child, peer), depending on link type. Link type depends on mutual agreements between the organizations operating the ASes at each end of each link.
+- Neighbor ISD-AS number
+- Neighbor interface underlay address
+
+In addition, each router must be configured with a list of interface IDs in use within the AS, containing:
 
-**Note:** The current SCION implementation uses the UDP/IP protocol as underlay. However, the use of other underlay protocols is possible and allowed.
+- Interface ID
+- "intra-protocol" address of the remote router
```

Comment:
Text does not make it quite clear which border routers get which list of values.

I'm guessing that it means to say that routers that "implement"/"handle" a particular interface need to know the link type/neighbor ISD-AS/neighbor interface address, while all other routers only need the address of the router that handles the Iface.

I believe all routers need to know the link type, to validate segment cross over (e.g. cross-over from an peer- to a core- link would be forbidden). Unfortunately, this is not described in the "processing at routers" section at all. Notably this _is_ described in the book v2 (5.6.3 processing at routers). 
The neighbor's ISD-AS number is currently not used for much, but possibly it should be used to drop packets with bad source/dest address.


Maybe it's easier to combine the two "lists". Suggestion:
```
For each link to a neighbor, these following information needs to be configured for all border routers:
- Interface ID
- Link Type,
- Neighbor ISD-AS number,
- In the case of the border router responsible for this interface: the neighbor interface's underlay address.
- Otherwise: the "intra-protocol" address of the border router responsible this interface.
```

Reply:
> Maybe it's easier to combine the two "lists"

But routers within the same AS require less parameters. For example, AFAIK no link type or neighbor ISD-AS is needed.  
Also, is it a requirement that all BRs within an AS see each others as "neighbors", in a meshed way? In other words, must each router have all interface ID mappings for its AS (and L3 connectivity to all other routers)?

Reply:
These are implementation details, I would say. That section is an explanation about configuration; in response to a question. It is not actually mandating anything.

I rewote along the lines proposed my Matthias. We discuss that later.

Reply:
> But routers within the same AS require less parameters. For example, AFAIK no link type or neighbor ISD-AS is needed.

As I said above, they _do_ need to have the information about the link type.

>  Also, is it a requirement that all BRs within an AS see each others as "neighbors", in a meshed way? In other words, must each router have all interface ID mappings for its AS (and L3 connectivity to all other routers)?

Almost. The exact requirement is somewhat more nuanced. For the draft, I think this is good enough.

Just for the sake of completeness: the model of path segment crossover implicitly assumes that you can transit between any two child interfaces, and between any child and core interface. So, a router responsible for a child or core interface needs to know _at least_ all (other) child interfaces. All other hops (parent-child, core-core, child-peer), are announced explicitly; the corresponding connectivity only needs to exist if the hop is ever announced.